tests/Functional/DiscoveryTest.php

```diff
@@ -71,4 +73,35 @@ public function testNoDiscoveryFallbacks()
         $clientDefinition = $this->container->getDefinition('httplug.client.default');
         $this->assertEquals([HttpClientDiscovery::class, 'find'], $clientDefinition->getFactory());
     }
+
+    public function testEnableAutowiring()
+    {
+        $this->load([
+            'default_client_autowiring' => true,
+        ]);
+
+        $this->assertContainerBuilderHasService('httplug.client.default');
+        $this->assertContainerBuilderHasService('httplug.async_client.default');
+        $this->assertContainerBuilderHasAlias(HttpClient::class);
+        $this->assertContainerBuilderHasAlias(HttpAsyncClient::class);
+    }
+
+    public function testDisableAutowiring()
+    {
+        $this->load([
+            'default_client_autowiring' => false,
+        ]);
+
+        $this->assertContainerBuilderHasService('httplug.client.default');
+        $this->assertContainerBuilderHasService('httplug.async_client.default');
+
+        self::assertThat(
+            $this->container,
+            new LogicalNot(new ContainerBuilderHasAliasConstraint(HttpClient::class))
```

Comment:
i am afraid that this does not work on php 5.x because we use phpunit 5.7 then which is missing the LogicalNot class.

however, i would suggest to simply skip this test if the php version is < 7 rather than do something super complicated to test with end-of-life versions of php.

Reply:
Fixed.